src/rgw/rgw_perf_counters.cc

```diff
@@ -192,6 +202,56 @@ void tinc(const CountersContainer &counters, int idx, ceph::timespan amt) {
 
 } // namespace rgw::op_counters
 
+namespace rgw::persistent_topic_counters {
+
+PerfCounters* global_topic_counters = nullptr;
+
+const std::string rgw_topic_counters_key = "rgw_topic";
+
+PerfCounters* new_perf_counter(const std::string& name, CephContext *cct) {
+  PerfCountersBuilder pcb(cct, name, l_rgw_topic_first, l_rgw_topic_last);
+  add_rgw_topic_counters(&pcb);
+  PerfCounters* new_counters = pcb.create_perf_counters();
+  cct->get_perfcounters_collection()->add(new_counters);
+
+  return new_counters;
+}
+
+void global_topic_counters_init(CephContext *cct) {
+  global_topic_counters = new_perf_counter(rgw_topic_counters_key, cct);
+}
+
+
+CountersContainer::CountersContainer(const std::string& topic_name, CephContext *cct) {
+  const std::string topic_key = ceph::perf_counters::key_create(rgw_topic_counters_key, {{"Topic", topic_name}});
+  this->topic_counters = std::shared_ptr<PerfCounters>(new_perf_counter(topic_key, cct));
```

Comment:
The use of `this` is not necessary in all of the `CountersContainer` methods

Reply:
Ack!